scripts/ccf/member/add.sh

```diff
@@ -0,0 +1,87 @@
+#!/bin/bash
+
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+
+REPO_ROOT="$(realpath "$(dirname "$(realpath "${BASH_SOURCE[0]}")")/../../..")"
+
+ccf-member-add() {
+    set -e
+
+    source $REPO_ROOT/scripts/ccf/az-cleanroom-aci/setup.sh && az-cleanroom-aci-setup
+    source $REPO_ROOT/scripts/ccf/propose.sh
+    source $REPO_ROOT/scripts/ccf/member/info.sh
+    source $REPO_ROOT/scripts/ccf/member/id.sh
+    source $REPO_ROOT/scripts/ccf/member/use.sh
+
+    MEMBER_NAME=${1:-$MEMBER_NAME}
+    if [ -z "$MEMBER_NAME" ]; then
+        read -p "Enter member name: " MEMBER_NAME
+    fi
+    export MEMBER_NAME
+    USE_AKV=${USE_AKV:-false}
+
+    # Hopefully we will have a single command for this eventually, e.g.
+    #   az cleanroom governance member up $MEMBER_NAME
+
+    if [[ $USE_AKV == false ]]; then
+        # Generate the member identity
+        (cd $WORKSPACE && az cleanroom governance member keygenerator-sh | bash -s -- --name $MEMBER_NAME)
+    else
+        # Create a key in AKV and download the cert for proposing
+        az keyvault certificate create \
+            --vault-name $AKV_VAULT_NAME \
+            --name $MEMBER_NAME \
+            --policy "${AKV_POLICY:-$(cat $REPO_ROOT/scripts/akv/key_policy.json | jq -c .)}"
+        rm -rf $WORKSPACE/${MEMBER_NAME}_cert.pem
+        az keyvault certificate download \
+            --vault-name $AKV_VAULT_NAME \
+            --name $MEMBER_NAME \
+            --file $WORKSPACE/${MEMBER_NAME}_cert.pem
+    fi
+
+    # Propose adding the member to the network
+    MEMBER_CERT=$(jq -Rs . < $WORKSPACE/${MEMBER_NAME}_cert.pem) \
+        envsubst < $REPO_ROOT/governance/proposals/set_member.json | jq \
+            > $WORKSPACE/proposals/set_member_${MEMBER_NAME}.json
+    ccf-propose $WORKSPACE/proposals/set_member_${MEMBER_NAME}.json
+
+    # Check if the proposal is accepted, otherwise early out with instructions to wait
+    if [ `ccf-member-info | jq -r '.status'` != "Accepted" ]; then
+        if [ `ccf-member-info | jq -r '.status'` == "Open" ]; then
+            echo -e \
+            "Proposal submitted, await sufficient votes for members to be accepted\n" \
+            "Check member status with:\n" \
+            "$REPO_ROOT/scripts/ccf/member/info.sh"
+            return 0
+        else
+            return 1
+        fi
+    fi
+
+    (
```

Comment:
For what reason do we want subshell here?

Reply:
I had to remind myself so definitely worth a comment.

Added in https://github.com/microsoft/azure-privacy-sandbox-kms/pull/240/commits/637f2e21c90d6f8da5a34184dd142afb1ac97ce8